src-java/blue-green/src/main/java/org/openkilda/bluegreen/kafka/interceptors/VersioningInterceptorBase.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.bluegreen.kafka.interceptors;
+
+import static java.lang.String.format;
+
+import org.openkilda.bluegreen.BuildVersionObserver;
+import org.openkilda.bluegreen.ZkWatchDog;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+
+@Slf4j
+public abstract class VersioningInterceptorBase implements BuildVersionObserver {
+    public static final int VERSION_IS_NOT_SET_LOG_TIMEOUT = 60;
+    public static final int CANT_CONNECT_TO_ZOOKEEPER_LOG_TIMEOUT = 60;
+
+    protected String connectionString;
+    protected ZkWatchDog watchDog;
+    protected String componentName;
+    protected String runId;
+    protected Instant versionIsNotSetTimestamp = Instant.MIN;
+    protected Instant cantConnectToZooKeeperTimestamp = Instant.MIN;
+    protected volatile byte[] version;
+
+    protected boolean isVersionIsNotTimeoutPassed() {
```

Comment:
do not include `not` logic into a getter, better apply it it's result. At least it will fix this ugly name. Also, it can help to simplify the whole usage logic.

Reply:
renamed